### PR TITLE
Merge upstream 3.3.0

### PR DESCRIPTION
**Pull request by manicmaniac**
## Overview

- Merged upstream/3.3.0 into master at https://github.com/quipper/MessageKit/pull/23/commits/fba17512e63b31d2de3a46bc8307b8e9b94daed3
- Resolved some conflicts at https://github.com/quipper/MessageKit/pull/23/commits/1da92e70cabb86e8f0b88170d5e7bbd5ffc05475 and https://github.com/quipper/MessageKit/pull/23/commits/d3de8c328335e4172d2e604f70d5907c91aae321

All you need to see is https://github.com/quipper/MessageKit/pull/23/commits/1da92e70cabb86e8f0b88170d5e7bbd5ffc05475

## Screenshot

image

### CODE REVIEW

File: Sources/Controllers/MessagesViewController.swift [hunk 1]
```diff
@@ -478,7 +478,8 @@ UICollectionViewDelegateFlowLayout, UICollectionViewDataSource, UIGestureRecogni
     }
     
     @objc private func clearMemoryCache() {
-        MessageStyle.bubbleImageCache.removeAllObjects()
+        // Since quipper/MessageKit does not use image caches, the next line is commented out.
+        // MessageStyle.bubbleImageCache.removeAllObjects()
```
Comment (manicmaniac) [inline Sources/Controllers/MessagesViewController.swift:482]: As I mentioned in https://github.com/quipper/MessageKit/pull/22#discussion_r1304194334, `MessageStyle` enum is completely diverged from the original.
`MessageStyle.bubbleImageCache` does not exist in our version.

**Review by m-sugawara — APPROVED** [review]
LGTM 👍

**Review by elliekwon — APPROVED** [review]
LGTM! 🚀🚀

**Review by k-kohey — APPROVED** [review]
LGTM

**Pull request merged**